dasharo-stability/capsule-update.robot

```diff
@@ -0,0 +1,258 @@
+    ${original_bios_version}=    Extract BIOS Version    ${out}
+    Log To Console    \nBefore update ${original_bios_version}
+
+    ${out}=    Start Update Process    wrong_cert.cap
+    Should Not Contain    ${out}    (The platform will automatically reboot and disable Firmware Update Mode
+    Should Not Contain    ${out}    failed to query capsule capability
+    Should Contain    ${out}    CapsuleApp: creating capsule descriptors at
+    Should Contain    ${out}    :\\capsule_testing\\>
+
+    Set DUT Response Timeout    3m
+    ${out}=    Execute UEFI Shell Command    reset
+    Should Not Contain    ${out}    (The platform will automatically reboot and disable Firmware Update Mode
+
+    ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
+    Enter Submenu From Snapshot    ${boot_menu}    UEFI Shell
+    Read From Terminal Until    Shell>
+    ${out}=    Execute UEFI Shell Command    smbiosview -t 0
+    ${updated_bios_version}=    Extract BIOS Version    ${out}
+    Log To Console    \nAfter update ${updated_bios_version}
+
+    Enter Capsule Testing Folder
+    ${out}=    Execute UEFI Shell Command    CapsuleApp.efi -S
+    IF    '${CONFIG}' == 'qemu'
+        Should Contain    ${out}    Capsule Status: Security Violation
+    ELSE
+        Should Contain    ${out}    Capsule Status: Not Ready
+    END
+
+CUP002.001 Capsule Update With Wrong GUID
+    [Documentation]    This test aims to verify if the DUT will reject flashed using Capsule Update with capsule with invalid GUID.
+    Power On
+    ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
+    Enter Submenu From Snapshot    ${boot_menu}    UEFI Shell
+    Read From Terminal Until    Shell>
+
+    ${out}=    Execute UEFI Shell Command    smbiosview -t 0
+    ${original_bios_version}=    Extract BIOS Version    ${out}
+    Log To Console    \nBefore update ${original_bios_version}
+
+    ${out}=    Start Update Process    invalid_guid.cap
+    Should Not Contain    ${out}    (The platform will automatically reboot and disable Firmware Update Mode
+    Should Not Contain    ${out}    failed to query capsule capability
+    Should Contain    ${out}    CapsuleApp: creating capsule descriptors at
+    Should Contain    ${out}    :\\capsule_testing\\>
+
+    Set DUT Response Timeout    3m
+    ${out}=    Execute UEFI Shell Command    reset
+    Should Not Contain    ${out}    (The platform will automatically reboot and disable Firmware Update Mode
+
+    ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
+    Enter Submenu From Snapshot    ${boot_menu}    UEFI Shell
+    Read From Terminal Until    Shell>
+    ${out}=    Execute UEFI Shell Command    smbiosview -t 0
+    ${updated_bios_version}=    Extract BIOS Version    ${out}
+
+    Log To Console    \nAfter update ${updated_bios_version}
+    Should Be Equal    ${original_bios_version}    ${updated_bios_version}
+
+CUP050.001 Capsule Update
```

Comment:
Please add a comment above the case which explains why there is a gap, otherwise someone may try to "fix" it.

Reply:
I have added the explanation in: 14485bb43d47ef903bc8838bda1992a9ab96a010